### ISSUE TITLE
Can we set the instanceId to the current IP address and port.

### ISSUE THREAD

**Issue by catcherwong** [SteeltoeOSS/Discovery#45]
Hello, 

Can we set the instanceId to the **current** IP address and port without specify the instanceId in `appsettings.json`?

For example, a product service has two instance, and their IP address and port are `10.0.10.1:9000` and `10.0.10.2:9001`.

I want to set the instanceIds of them are  `10.0.10.1:9000` and `10.0.10.2:9001` automatically.

Java client can do this by the following configuration

```yml
eureka:
  instance:
    prefer-ip-address: true
    instance-id: ${spring.cloud.client.ip-address}:${server.port}
  client:
    serviceUrl:
      defaultZone: http://10.0.10.100/eureka
```

I take a look on the code, when we not specify the instanceId, it will set the instanceId like the following format.

```cs
options.InstanceId = options.GetHostName(false) + ":" + options.AppName + ":" + options.NonSecurePort;
```

Maybe we can set a format string to the instanceId ?

So that we can make it more flexible.

**Comment by dtillman** [comment]
@catcherwong : We have in our roadmap to add ${.....} support to all of the steeltoe packages .. so you can customize the settings like you are wanting to do.. we just haven't gotten around to it   ..

**Comment by jkonicki** [comment]
This was addressed in 2.2.0 release.
